Components/Queues/QueueProducer.php

```diff
@@ -69,10 +69,7 @@ public function send(Exchange $ex, EndpointInterface $endpoint)
         $this->beforeSend($queueMessage, $options);
 
         // Send
-        if (!$queueDriver->isConnected()) {
-            $queueDriver->connect();
-        }
-
+        $queueDriver->connect();
```

Comment:
With the current setup this means that a new stream is created each time a message is produced using the amqp driver because the driver doesn't have inside the connect any logic to verify if it is connected or not

Reply:
See the comment above.